sklearn/ensemble/_hist_gradient_boosting/tests/test_gradient_boosting.py

```diff
@@ -192,6 +192,24 @@ def test_least_absolute_deviation():
     assert gbdt.score(X, y) > .9
 
 
+def test_poisson_loss():
```

Comment:
Would it make sense to also test that the score is above a given threshold?

Reply:
Unlike r2 score, it is hard to give an absolute "good" value for the poisson deviance. With the "d2 score" of #15244 this would make more sense.
I added a `DummyRegressor` with mean as prediction. This is (almost) equivalent to a d2 score. And I added out-of-sample tests.